proposals/utils/stepper.py

```diff
@@ -66,7 +66,7 @@ def attachment_slots(
                     obj,
                     force_desiredness=desiredness.EXTRA,
                 )
-                success = empty_slot.match(exclude=exclude)
+                success = empty_slot.match_and_set(exclude=exclude)
```

Comment:
This is why `match_and_set()` needs to return something. Its return value matters